scripts/APPS-162-delete-emails.js

```diff
@@ -0,0 +1,315 @@
+								{ 
+									topicId: 
+										{
+											$in: topicIds
+										}
+								}
+							]
+								
+				}
+			)
+			
+			if(numRemainingInSubsUnconfirmedForRemainingSubscodesForSpecifiedTopicIds > 0) {
+				throw new Error(`Some logs still exist in ${coll} for subscodes related to the specified topics!`);
+			}
+		}
+		
+		
+	}
+		
+	let numOtherSubLogs = await db.collection("subs_logs").countDocuments(
+		{
+			$or: 
+				 [
+					  { confirmEmail: 
+						   { $elemMatch: 
+								{ topicId: {
+									 $nin: topicIds
+									 }
+								}
+						   }
+					  },
+					  { subsEmail: 
+						   { $elemMatch: 
+								{ topicId: {
+									 $nin: topicIds
+									 }
+								}
+						   }
+					  },
+					  { unsubsEmail: 
+						   { $elemMatch: 
+								{ topicId: {
+									 $nin: topicIds
+									 }
+								}
+						   }
+					  },
+					  { resendEmail: 
+						   { $elemMatch: 
+								{ topicId: {
+									 $nin: topicIds
+									 }
+								}
+						   }
+					  }
+					  
+				 ]
+		}
+	);
+	
+	await db.subs_logs.deleteMany({
+     $nor: 
+		 [
+			  { confirmEmail: 
+				   { $elemMatch: 
+						{ topicId: {
+							 $nin: topicIds
+							 }
+						}
+				   }
+			  },
+			  { subsEmail: 
+				   { $elemMatch: 
+						{ topicId: {
+							 $nin: topicIds
+							 }
+						}
+				   }
+			  },
+			  { unsubsEmail: 
+				   { $elemMatch: 
+						{ topicId: {
+							 $nin: topicIds
+							 }
+						}
+				   }
+			  },
+			  { resendEmail: 
+				   { $elemMatch: 
+						{ topicId: {
+							 $nin: topicIds
+							 }
+						}
+				   }
+			  }
+			  
+		 ]
+	});
+	
+	let arraysToPullFrom = ["confirmEmail", "subsEmail", "unsubsEmail", "resendEmail"];
+	for (let arr of arraysToPullFrom) {
+		await pull("subs_logs", topicIds, arr);
+	}
+	
+	let numSubLogs = await db.collection("subs_logs").countDocuments();
+	if(numSubLogs != numOtherSubLogs) {
+		throw new Error("We may have deleted too few or too many subs logs!");
+	}
+	
+	let numSubLogsWithSpecTopicIds = await db.collection("subs_logs").countDocuments(
+		{
+			$or: 
+				 [
+					  { confirmEmail: 
+						   { $elemMatch: 
+								{ topicId: {
+									 $in: topicIds
+									 }
+								}
+						   }
+					  },
+					  { subsEmail: 
+						   { $elemMatch: 
+								{ topicId: {
+									 $in: topicIds
+									 }
+								}
+						   }
+					  },
+					  { unsubsEmail: 
+						   { $elemMatch: 
+								{ topicId: {
+									 $in: topicIds
+									 }
+								}
+						   }
+					  },
+					  { resendEmail: 
+						   { $elemMatch: 
+								{ topicId: {
+									 $in: topicIds
+									 }
+								}
+						   }
+					  }
+					  
+				 ]
+		}
+	);
+	
+	if(numSubLogsWithSpecTopicIds > 0) {
+		throw new Error("We didn't delete all the sub logs for the specified topic IDs");
+	}
+	
+	
+}
+
+async function pull(coll, topicIds, arrayToPullFrom) {
+	await db.collection(coll).update(
+		{},
+		{ 
+			$pull:
+				{
+					arrayToPullFrom: 
```

Comment:
This don't set the variable, it only define the key to be `arrayToPullFrom`

And after I tried to fix that in my local test, it didn't updated the entry. I found out the option `multi: true` was needed.

So I suggest the following code for this function

```js
let updateQuery = { $pull: { } };
updateQuery[ "$pull"][ arrayToPullFrom ] = { topicId: { $in: topicIds } };

await db.collection( col1 ).update(
  {},
  updateQuery,
  { multi: true }
);
```

After that change, it was working as expected and I got the "DONE!" success message.

Reply:
I updated my pull function to match your logic.  Thanks @duboisp !

Reply:
Note: This is one of the complications of moving from Mongo Shell queries with values copy-and-pasted to a node script with dynamic values. 😛